r/src/array_from_vector.cpp

```diff
@@ -406,9 +406,12 @@ std::shared_ptr<Array> MakeFactorArray(Rcpp::IntegerVector_ factor,
     case Type::INT64:
       return MakeFactorArrayImpl<arrow::Int64Type>(factor, type);
     default:
-      Rcpp::stop(tfm::format("Cannot convert to dictionary with index_type %s",
-                             dict_type.index_type()->ToString()));
+      break;
   }
+
+  cpp11::stop("Cannot convert to dictionary with index_type '",
+              dict_type.index_type()->ToString().c_str(), "'");
```

Comment:
IIUC, `cpp11::stop` still uses printf-esque syntax
```suggestion
  cpp11::stop("Cannot convert to dictionary with index_type '%s'",
              dict_type.index_type()->ToString().c_str());
```

Reply:
oh yeah that's right, thanks